polars/polars-io/src/csv/write_impl.rs

```diff
@@ -265,9 +232,28 @@ pub(crate) fn write<W: Write>(
 
             let last_ptr = &col_iters[col_iters.len() - 1] as *const SeriesIter;
             let mut finished = false;
-            // loop rows
             while !finished {
+                // loop cols
                 for col in &mut col_iters {
+                    let datetime_format = match &options.datetime_format {
+                        Some(datetime_format) => Some(datetime_format.as_str()),
+                        None => match col.dtype {
+                            DataType::Datetime(TimeUnit::Milliseconds, tz) => match tz {
+                                Some(_) => Some("%FT%H:%M:%S.%3f%z"),
+                                None => Some("%FT%H:%M:%S.%3f"),
+                            },
+                            DataType::Datetime(TimeUnit::Microseconds, tz) => match tz {
+                                Some(_) => Some("%FT%H:%M:%S.%6f%z"),
+                                None => Some("%FT%H:%M:%S.%6f"),
+                            },
+                            DataType::Datetime(TimeUnit::Nanoseconds, tz) => match tz {
+                                Some(_) => Some("%FT%H:%M:%S.%9f%z"),
+                                None => Some("%FT%H:%M:%S.%9f"),
+                            },
+                            _ => None,
+                        },
+                    };
+                    // loop rows
```

Comment:
Hey @alexander-beedie - would this be acceptable (match the dtype in the outer loop over columns before looping over rows)?